monitoring_service.rst

```diff
@@ -2,145 +2,230 @@ Raiden Network Monitoring Service
+
+Non-frameability
+----------------
+
+MSs can put the clients channel deposit at risk by ignoring all client requests. But since a MS can’t force other MSs to
+ignore client requests, this can not be considered as framing. When only a single MS is monitoring the channel, the MS’s
+dispute intervention and the reward payment happen atomically inside the SC. In this case, no party can frame the other.
+
+When multiple MSs try to settle the same dispute, only the first one doing so receives a reward, but all of them have to
+invest resources to monitor the channel and spend gas to interact with the SC. If you find a way to continuously front
+run other MSs, you can drain their resources and block their only income. However, while doing so you fulfilled the MS’s
+duty to settle the payment channel correctly and protect the client’s deposit.
+In the short run, this is an acceptable outcome for the client. In the long run, this will drive other MSs out of
+business and thus reduce redundancy and reliability of the overall MS ecosystem. Since all MSs try to be the first to
+submit a BP, it is unlikely that a single MS will continuously be the fastest, but slightly slower MSs will still not
+get any rewards even if they are well behaved and reliable.
+
+If a client wants to waste the resources of MSs, he can first broadcast a BP with a high reward and keep more recent BPs
+to himself. When a dispute happens, he can wait for the MSs to act before submitting his latest BPs, which prevents the
+MSs from receiving a reward. Doing this at a large scale is expensive, since the client needs to open and close a
+payment channel for this at his own cost.
+
+Recourse as a financial deterrent
+---------------------------------
+
+There is no possibility of recourse which lets MSs operate without any incentive of high reliability. A client must
+expect MSs to ignore their requests and have no means to force a highly reliable monitoring.
+
+Efficiency Requirements
+-----------------------
+
+For each channel, only the latest (as indicated by the nonce) BP has to be saved. Unless an extremely high amount of
+channels is being monitored, this efficiency should not be a concern for the MS.
+A client can use a single deposit to request an MS to monitor all his payment channels. If this causes the MS to monitor
+a problematically high amount of channels, he can start to ignore requests made by this client, or even drop old
+requests. Since there is no punishment for failing to monitor a channel, stopping to monitor is a simple way to reduce
+resource usage when desired, although it should not be necessary under normal circumstances.
 
-How will raiden nodes specify/deposit the monitoring fee? How will it be collected?
+Proposed SC logic
+'''''''''''''''''
 
-A scheme to prevent unnecessary simultaneous updates needs to exist. Options:
-MS chose an order amongst themselves
+1)  Raiden node will transfer tokens used as a reward to the channel smart contract
```

Comment:
Is this correct? The token are already in the UDC, right?

Reply:
yes good catch